pkg/server/bootstrap/identity.go

```diff
@@ -157,26 +160,48 @@ func wildcardIdentitiesResolver(ids *identities,
 				continue
 			}
 
-			apiExport, err := kcpClient.ApisV1alpha1().APIExports().Get(ctx, name, metav1.GetOptions{})
+			apiExportIdentity, err := apiExportIdentityProviderFn(ctx, name)
 			if err != nil {
 				errs = append(errs, err)
 				continue
 			}
-			if apiExport.Status.IdentityHash == "" {
+			if apiExportIdentity == "" {
 				errs = append(errs, fmt.Errorf("APIExport %s is not ready", name))
 				continue
 			}
 
 			ids.lock.Lock()
-			ids.groupResourceIdentities[gr] = apiExport.Status.IdentityHash
+			ids.groupResourceIdentities[gr] = apiExportIdentity
 			ids.lock.Unlock()
 
-			klog.V(2).Infof("APIExport %s|%s for resource %s.%s has identity %s", logicalcluster.From(apiExport), name, gr.Resource, gr.Group, apiExport.Status.IdentityHash)
+			klog.V(2).Infof("APIExport %s|%s for resource %s.%s has identity %s", tenancyv1alpha1.RootCluster, name, gr.Resource, gr.Group, apiExportIdentity)
 		}
 		return errorsutil.NewAggregate(errs)
 	}
 }
 
+func apiExportIdentityProviderFn(kcpClient kcpclient.Interface, kubeClusterClient kubernetes.ClusterInterface) func(ctx context.Context, apiExportName string) (string, error) {
```

Comment:
nit: don't name this `Fn` - we can see it's a function today and it may not be one in the future. `apiExportIdentityResolver` or similar describes what it does

Reply:
while `kcpClient` and `kubeClusterClient` are technically not incorrect names, it would be nice to know that they are for different shards - perhaps `rootShardClient` and `localShardClient`?

Reply:
sure, np will rename